File: fundamentalsAlgo/Python/palindrome.py


# O(n) time and O(1) space
def palindrome(s):
    # s=list(s)
    l,r=0,len(s)-1
    while l <= r:
        if s[l].lower()!=s[r].lower():
            print('check')
            return "Not a Palindrome"
        l+=1
        r-=1
    return "It is a Palindrome"

out=palindrome("bubba")
print(out)

File: fundamentalsAlgo/JavaScript/palindrome.js
// O(n) time and O(1) space
function palindrome(s) {
  let left = 0;
  let right = s.length - 1;
  while (left <= right) {
    if (s[left].toLowerCase() !== s[right].toLowerCase()) {
      return "Not a Palindrome";
    }
    left += 1;
    right -= 1;
  }
  return "It is a Palindrome";
}

const out = palindrome("bubba");
console.log(out);

module.exports = { palindrome };
